internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -118,6 +130,34 @@ func (hr *HypershiftAgentServiceConfigReconciler) getKubeconfigSecret(ctx contex
 	return secret, nil
 }
 
+func (hr *HypershiftAgentServiceConfigReconciler) ensureSpokeAgentInstallCRDs(ctx context.Context) error {
+	// Fetch agent-install CRDs using in-cluster client
+	localCRDs := &apiextensionsv1.CustomResourceDefinitionList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabels{fmt.Sprintf("operators.coreos.com/assisted-service-operator.%s", hr.Namespace): ""},
+	}
+	if err := hr.Client.List(ctx, localCRDs, listOpts...); err != nil {
+		return pkgerror.Wrap(err, "Failed to list in-cluster CRDs")
+	}
+	if len(localCRDs.Items) == 0 {
+		return pkgerror.New("agent-install CRDs are not available")
+	}
+
+	// Ensure CRDs exist on the spoke cluster
+	for _, crd := range localCRDs.Items {
+		c := crd
+		c.ResourceVersion = "" // ResourceVersion should not be set on objects to be created
+		result, err := controllerutil.CreateOrUpdate(ctx, hr.SpokeClient, &c, func() error { return nil })
```

Comment:
This won't update anything I think.
We'd need to set the spec to what we want it to be in the mutate function.

Reply:
Not sure what we want to update actually, maybe just the spec.